lib/os/mpsc_pbuf.c

```diff
@@ -576,6 +582,11 @@ const union mpsc_pbuf_generic *mpsc_pbuf_claim(struct mpsc_pbuf_buffer *buffer)
 		k_spin_unlock(&buffer->lock, key);
 	} while (cont);
 
+#ifdef CONFIG_MULTITHREADING
+	if (need_post && item == NULL) {
+		k_sem_give(&buffer->sem);
+	}
+#endif
```

Comment:
Please use IS_ENABLED so that remaining ifdefs can be removed.

```suggestion
	if (IS_ENABLED(CONFIG_MULTITHREADING) && need_post && item == NULL) {
		k_sem_give(&buffer->sem);
	}
```